tests/utils/test_number_conversion.py

```diff
@@ -4,22 +4,29 @@
 
 
 @pytest.mark.parametrize(
-    "num_ranks,global_num_samples,expected",
-    [(2, 10, 5), (2, 11, 5)],
+    "num_ranks,global_num_samples,local_micro_batch_size,expected",
+    [(2, 100, 10, 5), (2, 110, 10, 5)],
 )
-def test_get_local_num_batches_from_num_samples(num_ranks: int, global_num_samples: int, expected: int):
-    assert NumberConversion.get_local_num_batches_from_num_samples(num_ranks, global_num_samples) == expected
+def test_get_local_num_batches_from_num_samples(
+    num_ranks: int, global_num_samples: int, local_micro_batch_size: int, expected: int
+):
+    assert (
+        NumberConversion.get_local_num_batches_from_num_samples(num_ranks, global_num_samples, local_micro_batch_size)
+        == expected
+    )
 
 
 @pytest.mark.parametrize(
-    "num_ranks,global_num_tokens,sequence_length,expected",
-    [(2, 10, 2, 2), (2, 11, 2, 2), (2, 12, 2, 3)],
+    "num_ranks,global_num_tokens,sequence_length,local_micro_batch_size,expected",
+    [(2, 100, 2, 10, 2), (2, 110, 2, 10, 2), (2, 120, 2, 10, 3)],
```

Comment:
```suggestion
    [(2, 100, 2, 10, 2), (2, 110, 2, 10, 2), (2, 120, 2, 10, 3), (4, 100, 3, 4, 2)],
```
might be good to add more examples, with `num_ranks != 2` and `sequence_length != 2`